### ISSUE TITLE
How did you determine the details of the network parameters?

### ISSUE THREAD

**Issue by fzh0917** [huanglianghua/siamrpn-pytorch#5]
In your code, the numbers of kernels of five convolutional layers are `192, 512, 768, 768, 512`, respectively. How did you get these numbers? 
In the SiamRPN paper, the authors say that they use the AlexNet for extracting features of template branch and detection branch, but the numbers of convolutional kernels of the AlexNet isn't same as yours. So, I am confused. 
I am looking forward to your reply, thank you!

**Comment by fzh0917** [comment]
Beside, I have some other problems:
1. how do get stride and padding numbers of each layer? 
2. How do you know if each layer of the network needs `MaxPooling`, `ReLU` and `BatchNormalization`?

**Comment by Singapore-mor** [comment]
> Beside, I have some other problems:
> 
> 1. how do get stride and padding numbers of each layer?
> 2. How do you know if each layer of the network needs `MaxPooling`, `ReLU` and `BatchNormalization`?

I have  the same problems as you, in the essay the last kernels shoule be 256, while in this code is 512

**Comment by fzh0917** [comment]
> > Beside, I have some other problems:
> > 
> > 1. how do get stride and padding numbers of each layer?
> > 2. How do you know if each layer of the network needs `MaxPooling`, `ReLU` and `BatchNormalization`?
> 
> I have the same problems as you, in the essay the last kernels shoule be 256, while in this code is 512

I think that these numbers are the author(@huanglianghua)'s experiences, which make the performance of this reproduction better.

**Comment by Singapore-mor** [comment]
maybe you can find these parameters in pysot? I haven't seen it, maybe some details in pysot are more convincible?


| |
zxfans
|
|
邮箱：zhfans@126.com
|

Signature is customized by Netease Mail Master

On 10/20/2019 19:25, Zhihong Fu wrote:

Beside, I have some other problems:

how do get stride and padding numbers of each layer?
How do you know if each layer of the network needs MaxPooling, ReLU and BatchNormalization?

I have the same problems as you, in the essay the last kernels shoule be 256, while in this code is 512

I think that these numbers are the author(@huanglianghua)'s experiences, which make the performance of this reproduction better.

**Comment by fzh0917** [comment]
> maybe you can find these parameters in pysot? I haven't seen it, maybe some details in pysot are more convincible? 

No, absolutely convincible.
